docs/src/tutorials/homog/warp.md

```diff
@@ -0,0 +1,69 @@
+# Tutorials
+
+## Homography warps
+```@contents
+Pages = ["warp.md"]
+Depth = 5
+```
+Homography matrices are projective transformations. They can be represented using `DiffImages.Homography()` in Julia using `DiffImages`.
+
+In this example, we will train a homography matrix using DiffImages.jl.
+### Importing the libraries
+```@repl 1
+using DiffImages, ImageCore, ImageTransformations, FileIO, Zygote
+```
+
+### Loading the images
+Let us load the images first. We will also convert them to `Float32` precision type since we do not need such high precision.
+```@repl 1
+img = load("source.jpg") .|> RGB{Float32} 
+tgt = load("target.jpg") .|> RGB{Float32}
+```
+| Source Image | Destination Image |
+|--------------|-------------------|
+| ![src](source.jpg) | ![tgt](target.jpg) |
+
+### Initializing the matrix and hyperparameters
+Now let us define the homography matrix and other parameters such as the learning rate.
+```@repl 1
+h = DiffImages.Homography{Float32}()
+η = 2e-10 # Varies a lot example to example
+num_iters = 100
+```
+### Defining the criterion
+Nice! Now before we jump to the training loop, let us first define an `Images`-centric version of the mean squared error loss as our criterion.
+```@repl 1
+function image_mse(y, ŷ)
+    l = map((x, y) -> (x - y), y, ŷ)
+    l = mapreducec.(x->x^2, +, 0, l)
+    l = sum(l)
+    l
+end
+```
+
+### Defining the training loop
+Great! Now that we have defined our criterion, let us now define the training loop.
+```julia
+for i in 1:num_iters
+    ∇H, = Zygote.gradient(h) do trfm
+            out = ImageTransformations.warp(img, trfm, axes(img), zero(eltype(img)))
+            out = image_mse(out, tgt)
+            out
+        end
+
+    out = ImageTransformations.warp(img, h, axes(img), zero(eltype(img)))
+    println("Iteration: $i Loss: $(image_mse(out, tgt))")
+
+    h = h.H - η * (∇H.H)
+    h = DiffImages.Homography(h |> SMatrix{3, 3, Float32, 9})
+end
+```
```

Comment:
The issue here is that even with all previous blocks defined as `@repl`, none of those codes are actually running; they only import or define some functions without actually running.

So maybe actually run one iteration, and explain what `∇H` is in this scenario, and then display the `1:num_iters` results.

Reply:
Sure. I added the mathematical representation of ∇H, and also the outputs of the training loops in both of them